addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/internal/handler/FeedHandler.java

```diff
@@ -117,56 +117,62 @@ private void refreshFeedState() {
 
     private void publishChannelIfLinked(ChannelUID channelUID) {
         String channelID = channelUID.getId();
-        if (currentFeedState != null) {
-            if (isLinked(channelID)) {
-                State state = null;
-                switch (channelID) {
-                    case CHANNEL_LATEST_TITLE:
-                        String title = getLatestEntry(currentFeedState).getTitle();
-                        state = new StringType(getValueSafely(title));
-                        break;
-                    case CHANNEL_LATEST_DESCRIPTION:
-                        String description = getLatestEntry(currentFeedState).getDescription().getValue();
-                        state = new StringType(getValueSafely(description));
-                        break;
-                    case CHANNEL_LATEST_PUBLISHED_DATE:
-                        Date date = getLatestEntry(currentFeedState).getPublishedDate();
-                        Calendar calender = new GregorianCalendar();
-                        calender.setTime(date);
-                        state = new DateTimeType(calender);
-                        break;
-                    case CHANNEL_AUTHOR:
-                        String author = currentFeedState.getAuthor();
-                        state = new StringType(getValueSafely(author));
-                        break;
-                    case CHANNEL_DESCRIPTION:
-                        String channelDescription = currentFeedState.getDescription();
-                        state = new StringType(getValueSafely(channelDescription));
-                        break;
-                    case CHANNEL_TITLE:
-                        String channelTitle = currentFeedState.getTitle();
-                        state = new StringType(getValueSafely(channelTitle));
-                        break;
-                    case CHANNEL_LAST_UPDATE:
-                        Date pubDate = currentFeedState.getPublishedDate();
-                        Calendar calendar = new GregorianCalendar();
-                        calendar.setTime(pubDate);
-                        state = new DateTimeType(calendar);
-                        break;
-                    case CHANNEL_NUMBER_OF_ENTRIES:
-                        int numberOfEntries = currentFeedState.getEntries().size();
-                        state = new DecimalType(numberOfEntries);
-                        break;
-                }
-                if (state != null) {
-                    updateState(channelID, state);
-                } else {
-                    logger.debug("Can not update channel with ID : {} - channel name might be wrong!", channelID);
+
+        if (currentFeedState == null) {
+            // This will happen if the binding could not download data from the server
+            logger.info("Cannot update channel with ID {}; no data has been downloaded from the server!", channelID);
+            return;
+        }
+
+        if (!isLinked(channelID)) {
```

Comment:
I would suggest to use `isLinked` with channelUID for performance reasons.

Reply:
Done.